### PR TITLE
ticdc: add ticdc to pulsar doc

### PR DESCRIPTION
**Pull request by hfxsd**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/14913
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dumpling-overview.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ Compared to Mydumper, Dumpling has the following improvements:
 
 ### Required privileges
 
+- PROCESS: Required to query the cluster information to obtain the PD address and then control GC via the PD.
```
Comment (hfxsd) [inline dumpling-overview.md:77]:
```suggestion
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+  {matcher = ['test1.*', 'test5.*'], topic = "Topic expression 3", partition = "table"},
+  {matcher = ['test6.*'], partition = "default"},
+  {matcher = ['test7.*'], partition = "test123"}
+]
+```
+
+- The tables that match the matcher rule are dispatched according to the policy specified by the corresponding topic expression. For example, the table `test3.aa` is dispatched according to `Topic expression 2`, and the table `test5.aa` is dispatched according to `Topic expression 3`.
+- For a table that matches more than one matcher rule, the topic expression corresponding to the top matcher will take precedence. For example, the table `test1.aa` is dispatched according to `Topic expression 1`.
+- For tables that do not match any matcher, the corresponding data change events are sent to the default topic specified in `-sink-uri`. For example, the table `test10.aa` is sent to the default topic.
+- For tables that match the matcher rule but do not have a topic dispatcher specified, the corresponding data changes are sent to the default topic specified in `-sink-uri`. For example, the table `test9.abc` is sent to the default topic.
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-pulsar.md:238]:
```suggestion
- For tables that match the matcher rule but do not have a topic dispatcher specified, the corresponding data changes are sent to the default topic specified in `-sink-uri`. For example, the table `test6.abc` is sent to the default topic.
```

File: ticdc/ticdc-changefeed-config.md [hunk 3]
```diff
@@ -224,4 +227,45 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+oauth2.oauth2-scope="xxxx"
+# The number of cached Pulsar producers in TiCDC. The value is 10240 by default. Each Pulsar producer corresponds to one topic. If you need to replicate more than the default number of topics, you need to increase the number.
+pulsar-producer-cache-size=10240
+# Pulsar data compression method. No compression is used by default. Optional values are "lz4", "zlib", and "zstd".
+compression-type=""
+# The timeout for the Pulsar client to establish a TCP connection with the server. The value is 5 seconds by default.
+connection-timeout=5
+# The timeout for Pulsar clients to initiate operations such as create and subscribe. The value is 30 seconds by default.
+operation-timeout=30
+# The maximum number of messages in a single batch for a Pulsar producer to send. The value is 1000 seconds by default.
```
Comment (asddongmen) [inline ticdc/ticdc-changefeed-config.md:265]:
```suggestion
# The maximum number of messages in a single batch for a Pulsar producer to send. The value is 1000 by default.
```

File: ticdc/ticdc-changefeed-config.md [hunk 4]
```diff
@@ -224,4 +227,45 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+oauth2.oauth2-client-id="0Xx...Yyxeny"
+# Pulsar oauth2 oauth2-scope
+oauth2.oauth2-scope="xxxx"
+# The number of cached Pulsar producers in TiCDC. The value is 10240 by default. Each Pulsar producer corresponds to one topic. If you need to replicate more than the default number of topics, you need to increase the number.
+pulsar-producer-cache-size=10240
+# Pulsar data compression method. No compression is used by default. Optional values are "lz4", "zlib", and "zstd".
+compression-type=""
+# The timeout for the Pulsar client to establish a TCP connection with the server. The value is 5 seconds by default.
+connection-timeout=5
+# The timeout for Pulsar clients to initiate operations such as create and subscribe. The value is 30 seconds by default.
```
Comment (asddongmen) [inline ticdc/ticdc-changefeed-config.md:263]:
```suggestion
# The timeout for Pulsar clients to initiate operations such as create and subscribe a topic. The value is 30 seconds by default.
```

**Review by asddongmen — APPROVED** [review]

File: ticdc/ticdc-changefeed-config.md [hunk 5]
```diff
@@ -224,4 +227,45 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+token-from-file="/data/pulsar/token-file.txt"
+# Pulsar uses the basic account and password to authenticate the identity. Specify the account.
+basic-user-name="root"
+# Pulsar uses the basic account and password to authenticate the identity. Specify the password.
+basic-password="password"
+# The certificate path for Pulsar TLS encrypted authentication.
+auth-tls-certificate-path="/data/pulsar/certificate"
+# The private key path for Pulsar TLS encrypted authentication.
+auth-tls-private-key-path="/data/pulsar/certificate.key"
+# The credential file path for Pulsar TLS encrypted authentication.
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:245]:
```suggestion
# Path to trusted certificate file of the Pulsar TLS encrypted authentication.
```

Comment (qiancai) [reply ticdc/ticdc-changefeed-config.md:245]: 和 ticdc/ticdc-sink-to-pulsar.md 里的翻译保持一致

File: ticdc/ticdc-sink-to-pulsar.md [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+token-from-file="/data/pulsar/token-file.txt"
+# Pulsar uses the basic account and password to authenticate the identity. Specify the account.
+basic-user-name="root"
+# Pulsar uses the basic account and password to authenticate the identity. Specify the password.
+basic-password="password"
+# The certificate path for Pulsar TLS encrypted authentication.
+auth-tls-certificate-path="/data/pulsar/certificate"
+# The private key path for Pulsar TLS encrypted authentication.
+auth-tls-private-key-path="/data/pulsar/certificate.key"
+# The credential file path for Pulsar TLS encrypted authentication.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:106]:
```suggestion
# Path to trusted certificate file of the Pulsar TLS encrypted authentication.
```

Comment (qiancai) [reply ticdc/ticdc-sink-to-pulsar.md:106]: 和 ticdc/ticdc-sink-to-pulsar.md 里的翻译保持一致

File: ticdc/ticdc-sink-to-pulsar.md [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+dispatchers = [
+  {matcher = ['test1.*', 'test2.*'], topic = "Topic expression 1", partition = "ts" },
+  {matcher = ['test3.*', 'test4.*'], topic = "Topic expression 2", partition = "index-value" },
+  {matcher = ['test1.*', 'test5.*'], topic = "Topic expression 3", partition = "table"},
+  {matcher = ['test6.*'], partition = "default"},
+  {matcher = ['test7.*'], partition = "test123"}
+]
+```
+
+- The tables that match the matcher rule are dispatched according to the policy specified by the corresponding topic expression. For example, the table `test3.aa` is dispatched according to `Topic expression 2`, and the table `test5.aa` is dispatched according to `Topic expression 3`.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:235]:
```suggestion
- The tables that match a matcher rule are dispatched according to the policy specified by the corresponding topic expression. For example, the table `test3.aa` is dispatched according to `Topic expression 2`, and the table `test5.aa` is dispatched according to `Topic expression 3`.
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+- For a table that matches more than one matcher rule, the topic expression corresponding to the top matcher will take precedence. For example, the table `test1.aa` is dispatched according to `Topic expression 1`.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:236]:
```suggestion
- For a table that matches more than one matcher rule, it is dispatched according to the first matching topic expression. For example, the table `test1.aa` is dispatched according to `Topic expression 1`.
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+- For tables that do not match any matcher, the corresponding data change events are sent to the default topic specified in `-sink-uri`. For example, the table `test10.aa` is sent to the default topic.
+- For tables that match the matcher rule but do not have a topic dispatcher specified, the corresponding data changes are sent to the default topic specified in `-sink-uri`. For example, the table `test6.abc` is sent to the default topic.
+
+### Topic dispatcher
+
+A topic dispatcher is specified with `topic = "xxx"` and uses topic expressions to implement a flexible topic dispatching policy. It is recommended that the total number of topics be less than 1000.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:242]:
```suggestion
You can use `topic = "xxx"` to specify a topic dispatcher and use topic expressions to implement flexible topic dispatching policies. It is recommended that the total number of topics be less than 1000.
```

Comment (qiancai) [reply ticdc/ticdc-sink-to-pulsar.md:242]: align with the translation in https://docs.pingcap.com/tidb/dev/ticdc-sink-to-kafka#topic-dispatchers

File: ticdc/ticdc-sink-to-pulsar.md [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+
+The basic pattern of a topic expression is `[prefix]{schema}[middle][{table}][suffix]`. The following are the meanings of each part:
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:244]:
```suggestion
The format of a topic expression is `[prefix]{schema}[middle][{table}][suffix]`. The following are the meanings of each part:
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+
+- `prefix`: Optional. Represents the prefix of the topic name.
+- `{schema}`: Optional. Represents the database name.
+- `middle`: Optional. Represents the separator between a database and a table.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:248]:
```suggestion
- `middle`: Optional. Represents the separator between a database name and a table name.
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+- `matcher = ['test1.table1', 'test2.table2'], topic = "hello_{schema}_{table}"`
+    - Data change events corresponding to the table `test1.table1` are despatched to a topic named `hello_test1_table1`.
+    - Data change events corresponding to the table `test2.table2` are despatched to a topic named `hello_test2_table2`.
+
+- `matcher = ['test3.*', 'test4.*'], topic = "hello_{schema}_world"`
+    - Data change events for all tables under `test3` are despatched to a topic named `hello_test3_world`.
+    - Data change events for all tables under `test4` are despatched to a topic named `hello_test4_world`.
+
+- `matcher = ['*.*'], topic = "{schema}_{table}"`
+    - For all tables that TiCDC listens on, they are despatched to separate topics according to the `databaseName_tableName` rule. For example, for table `test.account`, TiCDC despatches its data change log to a topic named `test_account`.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:265]:
```suggestion
    - For all tables that TiCDC listens on, they are despatched to separate topics according to the `databaseName_tableName` rule. For example, for the table `test.account`, TiCDC despatches its data change log to a topic named `test_account`.
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+
+DDL statements such as `CREATE DATABASE` and `DROP DATABASE` that are not related to a specific table are called database-level DDL statements. Events corresponding to database-level DDL statements are dispatched to the default topic specified in `--sink-uri`.
+
+#### Table-level DDL events
+
+DDL statements such as `ALTER TABLE` and `CREATE TABLE` that are related to a specific table are called table-level DDL statements. Events corresponding to table-level DDL statements are dispatched to an appropriate topic according to the configuration of `dispatchers`.
+
+For example, for a `dispatchers` configuration like `matcher = ['test.*'], topic = {schema}_{table}`, the DDL events are despatched as follows:
+
+- If a single table is involved in the DDL event, the DDL event is dispatched to the appropriate topic as is. For example, for the DDL event `DROP TABLE test.table1`, the event is dispatched to the topic named `test_table1`.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:279]:
```suggestion
- If a DDL event only involves a single table, the DDL event is dispatched to the appropriate topic as it is. For example, for the DDL event `DROP TABLE test.table1`, the event is dispatched to the topic named `test_table1`.
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+
+- If the DDL event involves more than one table (`RENAME TABLE`, `DROP TABLE`, and `DROP VIEW` might all involve more than one table), the single DDL event is split into multiple ones and dispatched to appropriate topics. For example, for the DDL event `RENAME TABLE test.table1 TO test.table10, test.table2 TO test.table20`, the processing is as follows:
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:281]:
```suggestion
- If a DDL event involves more than one table (`RENAME TABLE`, `DROP TABLE`, and `DROP VIEW` might all involve more than one table), the single DDL event is split into multiple ones and dispatched to appropriate topics. For example, for the DDL event `RENAME TABLE test.table1 TO test.table10, test.table2 TO test.table20`, the processing is as follows:
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+
+    - Dispatch the DDL event for `RENAME TABLE test.table1 TO test.table10` to a topic named `test_table1`.
+    - Dispatch the DDL event for `RENAME TABLE test.table2 TO test.table20` to a topic named `test_table2`.
+
+### Partition dispatcher
+
+Currently TiCDC only supports consumers to consume messages using the Exclusive subscription model, that is, each consumer can consume messages from all partitions in a topic.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:288]:
```suggestion
Currently, TiCDC only supports consumers to consume messages using the exclusive subscription model, that is, each consumer can consume messages from all partitions in a topic.
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+
+You can specify a partition dispatcher with `partition = "xxx"`. The following partition dispatches are supported: `default`, `ts`, `index-value`, and `table`. If you fill in any other string, it will be passed through to the `key` of the message in the message sent to the Pulsar server.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:290]:
```suggestion
You can specify a partition dispatcher with `partition = "xxx"`. The following partition dispatches are supported: `default`, `ts`, `index-value`, and `table`. If you fill in any other string, TiCDC will pass that string as the `key` of the message in the messages sent to the Pulsar server.
```

File: ticdc/ticdc-changefeed-config.md [hunk 17]
```diff
@@ -224,4 +227,45 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+oauth2.oauth2-issuer-url="https://xxxx.auth0.com"
+# Pulsar oauth2 audience
+oauth2.oauth2-audience="https://xxxx.auth0.com/api/v2/"
+# Pulsar oauth2 private-key
+oauth2.oauth2-private-key="/data/pulsar/privateKey"
+# Pulsar oauth2 client-id
+oauth2.oauth2-client-id="0Xx...Yyxeny"
+# Pulsar oauth2 oauth2-scope
+oauth2.oauth2-scope="xxxx"
+# The number of cached Pulsar producers in TiCDC. The value is 10240 by default. Each Pulsar producer corresponds to one topic. If you need to replicate more than the default number of topics, you need to increase the number.
```
[the remark this answers is not in the document]
Comment (qiancai) [reply ticdc/ticdc-changefeed-config.md:257]: 和 ticdc/ticdc-sink-to-pulsar.md 里的翻译保持一致

File: ticdc/ticdc-sink-to-pulsar.md [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+# dispatchers = [
+#    {matcher = ['test1.*', 'test2.*'], topic = "Topic expression 1", partition = "ts" },
+#    {matcher = ['test3.*', 'test4.*'], topic = "Topic expression 2", partition = "index-value" },
+#    {matcher = ['test1.*', 'test5.*'], topic = "Topic expression 3", partition = "table"},
+#    {matcher = ['test6.*'], partition = "default"},
+#    {matcher = ['test7.*'], partition = "test123"}
+# ]
+
+# `protocol` is used to specify the format protocol for encoding messages.
+# When the downstream is Pulsar, only the canal-json protocol is supported.
```
[the remark this answers is not in the document]
Comment (qiancai) [reply ticdc/ticdc-sink-to-pulsar.md:89]: 和其他地方的翻译保持一致

File: ticdc/ticdc-changefeed-config.md [hunk 19]
```diff
@@ -224,4 +227,45 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
 sasl-oauth-grant-type = "client_credentials"
 # The audience in the Kafka SASL OAUTHBEARER authentication. The default value is empty. This parameter is optional when the OAUTHBEARER authentication is used.
 sasl-oauth-audience = "kafka"
+
+# The following parameters take effect only when the downstream is Pulsar.
+[sink.pulsar-config]
+# Authentication on the Pulsar server side is done using a token. Specify the value of the token.
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:233]:
```suggestion
# Authentication on the Pulsar server is done using a token. Specify the value of the token.
```

File: ticdc/ticdc-changefeed-config.md [hunk 20]
```diff
@@ -224,4 +227,45 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+authentication-token = "xxxxxxxxxxxxx"
+# When you use a token for Pulsar server-side authentication, specify the path to the file where the token is located.
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:235]:
```suggestion
# When you use a token for Pulsar server authentication, specify the path to the file where the token is located.
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 21]
```diff
@@ -0,0 +1,298 @@
...
+oauth2.oauth2-issuer-url="https://xxxx.auth0.com"
+# Pulsar oauth2 audience
+oauth2.oauth2-audience="https://xxxx.auth0.com/api/v2/"
+# Pulsar oauth2 private-key
+oauth2.oauth2-private-key="/data/pulsar/privateKey"
+# Pulsar oauth2 client-id
+oauth2.oauth2-client-id="0Xx...Yyxeny"
+# Pulsar oauth2 oauth2-scope
+oauth2.oauth2-scope="xxxx"
+# The number of cached Pulsar producers in TiCDC. The value is 10240 by default. Each Pulsar producer corresponds to one topic. If you need to replicate more than the default number of topics, you need to increase the number.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:118]:
```suggestion
# The number of cached Pulsar producers in TiCDC. The value is 10240 by default. Each Pulsar producer corresponds to one topic. If the number of topics you need to replicate is larger than the default value, you need to increase the number.
```

Comment (qiancai) [reply ticdc/ticdc-sink-to-pulsar.md:118]: 和 ticdc/ticdc-sink-to-pulsar.md 里的翻译保持一致

File: ticdc/ticdc-changefeed-config.md [hunk 22]
```diff
@@ -224,4 +227,45 @@ sasl-oauth-scopes = ["producer.kafka", "consumer.kafka"]
...
+oauth2.oauth2-client-id="0Xx...Yyxeny"
+# Pulsar oauth2 oauth2-scope
+oauth2.oauth2-scope="xxxx"
+# The number of cached Pulsar producers in TiCDC. The value is 10240 by default. Each Pulsar producer corresponds to one topic. If you need to replicate more than the default number of topics, you need to increase the number.
+pulsar-producer-cache-size=10240
+# Pulsar data compression method. No compression is used by default. Optional values are "lz4", "zlib", and "zstd".
+compression-type=""
+# The timeout for the Pulsar client to establish a TCP connection with the server. The value is 5 seconds by default.
+connection-timeout=5
+# The timeout for Pulsar clients to initiate operations such as creating and subscribing a topic. The value is 30 seconds by default.
```
Comment (qiancai) [inline ticdc/ticdc-changefeed-config.md:263]:
```suggestion
# The timeout for Pulsar clients to initiate operations such as creating and subscribing to a topic. The value is 30 seconds by default.
```

File: ticdc/ticdc-sink-to-pulsar.md [hunk 23]
```diff
@@ -0,0 +1,298 @@
...
+batching-max-messages=1000
+# The interval at which Pulsar producer messages are saved for batching. The value is 10 milliseconds by default.
+batching-max-publish-delay=10
+# The timeout for a Pulsar producer to send a message. The value is 30 seconds by default.
+send-timeout=30
+```
+
+### Best practice
+
+* You need to specify the `protocol` parameter when creating a changefeed. Currently only the `canal-json` protocol is supported for replicating data to Pulsar.
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-pulsar.md:136]:
```suggestion
* You need to specify the `protocol` parameter when creating a changefeed. Currently, only the `canal-json` protocol is supported for replicating data to Pulsar.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**